src/config/api.rs

```diff
@@ -0,0 +1,82 @@
+use super::Config;
+use serde::{Deserialize, Serialize};
+use std::net::{Ipv4Addr, SocketAddr};
+
+#[derive(Default, Debug, Deserialize, Serialize, PartialEq)]
+pub struct Options {
+    #[serde(default = "default_enabled")]
+    pub enabled: bool,
+
+    #[serde(default = "default_bind")]
+    pub bind: Option<SocketAddr>,
+
+    #[serde(default = "default_playground")]
+    pub playground: bool,
+}
```

Comment:
Would it make sense to use `#[derivative(Default)]` to have the same defaults for serde and within Rust?